lib/compress/index.js

```diff
@@ -3522,6 +3533,7 @@ const pure_prop_access_globals = new Set([
                 return true;
             }
             if (node instanceof AST_This && this instanceof AST_Arrow) {
+                // TODO check arguments too!
```

Comment:
Will you continue work on that in this PR?

Reply:
Not really, just a note for later. This is because `arguments` can be used inside arrow functions and it uses the closest non-arrow `arguments` just like it does with `this`.